mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/mapper/MapperImplementationGenerator.java

```diff
@@ -117,7 +117,7 @@ protected ClassName getPrincipalTypeName() {
         MethodSpec.constructorBuilder().addModifiers(Modifier.PUBLIC);
 
     GeneratedCodePatterns.addFinalFieldAndConstructorArgument(
-        ClassName.get(MapperContext.class), "context", classContents, constructorContents);
+        ClassName.get(DefaultMapperContext.class), "context", classContents, constructorContents);
```

Comment:
Wouldn't it limit extensibility of `MapperContext`? What if some pro user decides to provide own implementation of `MapperContext`? With this change, it seems that implementation is hard-coded now

Reply:
`MapperContext` instances are only created from generated code. In order to plug your own context implementation, you'd have to write your own processor extension, I think this is beyond where even very advanced users are willing to go.

Also the "custom state" feature introduced in this PR provides some level of customization of the context.